tests/lib.rs

```diff
@@ -0,0 +1,197 @@
+    assert_eq!(parsed, Ip::V6(address, None));
+
+    assert!(Ip::parse("10.0.0.1%1").is_err());
+    assert!(Ip::parse("10.0.0.1%eth0").is_err());
+    assert!(Ip::parse("FE80::C001:1DFF:FEE0:0%").is_err());
+    assert!(Ip::parse("FE80::C001:1DFF:FEE0:0% ").is_err());
+
+    let parsed = Ip::parse("192.168.10.1").unwrap();
+    let address = Ipv4Addr::new(192, 168, 10, 1);
+    assert_eq!(parsed, Ip::V4(address));
+}
+
+#[test]
+fn test_nameserver() {
+    assert_eq!(
+        parse_str("nameserver 127.0.0.1").nameservers[0],
+        ip("127.0.0.1")
+    );
+    assert_eq!(
+        parse_str("nameserver 127.0.0.1#comment").nameservers[0],
+        ip("127.0.0.1")
+    );
+    assert_eq!(
+        parse_str("nameserver 127.0.0.1;comment").nameservers[0],
+        ip("127.0.0.1")
+    );
+    assert_eq!(
+        parse_str("nameserver 127.0.0.1 # another comment").nameservers[0],
+        ip("127.0.0.1")
+    );
+    assert_eq!(
+        parse_str("nameserver 127.0.0.1  ; ").nameservers[0],
+        ip("127.0.0.1")
+    );
+    assert_eq!(parse_str("nameserver ::1").nameservers[0], ip("::1"));
+    assert_eq!(
+        parse_str("nameserver 2001:db8:85a3:8d3:1319:8a2e:370:7348").nameservers[0],
+        ip("2001:db8:85a3:8d3:1319:8a2e:370:7348")
+    );
+    assert_eq!(
+        parse_str("nameserver ::ffff:192.0.2.128").nameservers[0],
+        ip("::ffff:192.0.2.128")
+    );
+}
+
+fn parse_file<P: AsRef<Path>>(path: P) -> resolv_conf::Config {
+    let mut data = String::new();
+    let mut file = File::open(path).unwrap();
+    file.read_to_string(&mut data).unwrap();
+    resolv_conf::Config::parse(&data).unwrap()
+}
+
+#[test]
+fn test_parse_simple_conf() {
+    let mut config = resolv_conf::Config::new();
+    config.nameservers.push(Ip::V4(Ipv4Addr::new(8, 8, 8, 8)));
+    config.nameservers.push(Ip::V4(Ipv4Addr::new(8, 8, 4, 4)));
+    assert_eq!(config, parse_file("tests/resolv.conf-simple"));
+}
+
+#[test]
+fn test_parse_linux_conf() {
+    let mut config = resolv_conf::Config::new();
+    config.search.push("example.com".into());
+    config.search.push("sub.example.com".into());
+    config.nameservers = vec![
+        Ip::V6(
+            Ipv6Addr::new(0x2001, 0x4860, 0x4860, 0, 0, 0, 0, 0x8888),
+            None,
+        ),
+        Ip::V6(
+            Ipv6Addr::new(0x2001, 0x4860, 0x4860, 0, 0, 0, 0, 0x8844),
+            None,
+        ),
+        Ip::V4(Ipv4Addr::new(8, 8, 8, 8)),
+        Ip::V4(Ipv4Addr::new(8, 8, 4, 4)),
+    ];
+    config.ndots = 8;
+    config.timeout = 8;
+    config.attempts = 8;
+    config.rotate = true;
+    config.inet6 = true;
+    config.no_tld_query = true;
+    config.sortlist = vec![
+        Network::V4(
+            Ipv4Addr::new(130, 155, 160, 0),
+            Ipv4Addr::new(255, 255, 240, 0),
+        ),
+        // This fails currently
+        Network::V4(Ipv4Addr::new(130, 155, 0, 0), Ipv4Addr::new(255, 255, 0, 0)),
```

Comment:
This is the failing test. Will look into it tonight.

Reply:
If ["DNS and Bind"](https://books.google.com/books?id=HggtWI1ShvMC&pg=PT133&lpg=PT133&dq=sortlist+mask+resolv&source=bl&ots=PTKqTUTz0v&sig=KpUKVZDtUyK3i2nsIS0Qf69TedI&hl=en&sa=X&ved=0ahUKEwiBmd-eouXXAhVK0oMKHQ8BB1gQ6AEINTAC#v=onepage&q=sortlist%20mask%20resolv&f=false) is correct, the expected mask is `255.255.0.0`, so this is a bug.

> The argument after the slash is the subnet mask for the subnet in question. To prefer an entire network, you can omit the slash and the subnet mask
>  `sortlist 128.32.0.0`
> The resolver then assumes you mean the entire network 128.32/16. (The resolver derives the default unsubnetted net mask for the network from the first two bits of the IP address.)

I assume in the last sentence they meant "bytes" not "bits", it does not make sense otherwise.